File: Dsa-day239.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        self.count = 0 
        self.ans = ""   
        
        def dfs(cur: str):
            if len(cur) == n:
                self.count += 1
                if self.count == k:
                    self.ans = cur
                return
            
            for c in "abc":
                if not cur or cur[-1] != c:
                    dfs(cur + c)
                    if self.ans:
                        return
        
        dfs("")
        return self.ans

File: Dsa-day239.cpp
class Solution {
public:
    string getHappyString(int n, int k) {
        string ans = "";
        int count = 0;
        function<bool(string&)> dfs = [&](string &cur) -> bool {
            if (cur.size() == n) { 
                count++;
                if (count == k) {  
                    ans = cur;
                    return true;
                }
                return false;
            }
            for (char c : {'a', 'b', 'c'}) {
                if (cur.empty() || cur.back() != c) { 
                    cur.push_back(c);
                    if (dfs(cur)) return true;
                    cur.pop_back();  
                }
            }
            return false;
        };
        
        string cur;
        dfs(cur);
        return ans;
    }
};
